### PR TITLE
feat: use mangrove-strats and mangrove-core

### PR DESCRIPTION
**Pull request by peterMangrove**
This PR updates the dependancy from mangrove-core to the new mangrove-strats repo, where core and strats are split up.

With the new configuration in core, on how we determine what addresses to read, we can now in mangrove.js control where the mangrove-core and mangrove-strats addresses are. This way we can depend on mangrove-core and mangrove-strats npm packages, separately and still read the correct addresses. 

The changes needed for this new configuration to work are:
1. We depend on both mangrove-core and mangrove-strats npm packages
2.  I order to be able to depend on a github branch, then we need to run foundry setup before we run `yarn install`, because when we depend on a github branch, it tries to run prepack in order to pack the repo. But this requires foundry.
3. Added foundry.toml, with correct remappings, read/write access, etc.
4. addresses/deployed.backup folder added, for deployment files. Needed as some tests deploy contracts and write addresses to that folder. 
5. `mgvConfig.json` added to configure where addresses are ( The new configuration from core)
6. Added code to read mangrove-strats addresses in SDK
7.  When running forge script, we no longer set root, as we need root to be the root of mangrove.js. (This is where the `mgvConfig.json` are)
8. Typechain has to get abis from both the mangrove-strats and mangrove-core packages
9. For some reason 4 tests were now using a bit less gas and the tests has been updated to reflect that. 

You can find the changes to mangrove-core [here](https://github.com/mangrovedao/mangrove-core/pull/469/files)
And the changes to mangrove-strats [here](https://github.com/mangrovedao/mangrove-strats/pull/28/files)

⚠️ This PR is a POC. As we need to have actual released core and strat packages to depend on, before we merge this to develop.

### CODE REVIEW

**Review by espendk — COMMENTED** [review]
This feels a bit to hack'ish for my taste. I think we should take a step back and consider if there's a better approach.

Btw, we should remember to load strat addresses in `configuration.ts`.

File: src/util/runTypechain.ts [hunk 1]
```diff
@@ -31,14 +31,20 @@ async function main() {
     ).dir;
     const coreAbisDir = `${coreDir}/dist/mangrove-abis`;
 
+    // Get directory for mangrove-strats module's abis
+    const stratsDir = path.parse(
+      require.resolve("@mangrovedao/mangrove-strats")
+    ).dir;
+    const stratsAbisDir = `${stratsDir}/dist/mangrove-abis`;
```
[the remark this answers is not in the document]
Comment (peterMangrove) [reply src/util/runTypechain.ts]: Fixed

File: package.json [hunk 2]
```diff
@@ -61,7 +61,7 @@
   },
   "dependencies": {
     "@ethersproject/experimental": "^5.7.0",
-    "@mangrovedao/mangrove-core": "^1.5.7",
+    "@mangrovedao/mangrove-strats": "https://github.com/mangrovedao/mangrove-strats.git#83349b8dccead1f8f308607620e6a20364a00216",
```
[the remark this answers is not in the document]
Comment (peterMangrove) [reply package.json]: Done

File: src/util/deal.ts [hunk 3]
```diff
@@ -28,8 +26,8 @@ export async function deal(dealParams: {
     url: dealParams.url,
     env: env,
     provider: dealParams.provider,
-    script: "GetTokenDealSlot",
-    coreDir: CORE_DIR,
+    script:
+      "./node_modules/@mangrovedao/mangrove-core/script/lib/GetTokenDealSlot.sol:GetTokenDealSlot",
```
Comment (espendk) [inline src/util/deal.ts:30]: This looks a bit fishy - what if the package is installed in a different path than `./node_modules` ?

**Review by espendk — COMMENTED** [review]
Added one question, but lgtm otherwise

**Review by espendk — APPROVED** [review]

**Pull request merged**